File: 921_min_add_make_paranthesis_valid.cpp
int minAddToMakeValid(string s) {
        int lc=0, ans=0;
        for(int i=0;i<s.size();i++){
            if(s[i]=='(')
                lc++;
            if(s[i]==')'){
                if(lc>0)
                    lc--;
                else{
                    while(i<s.size() && s[i]==')'){
                        ans+=1;
                        i++;
                    }
                    i--;
                }
                    
            }
        }
        
        return ans+lc;
    }

File: py/921_min_add_make_paranthesis_valid.py
def minAddToMakeValid(self, s: str) -> int:
        lc: int = 0
        extra: int = 0
        
        for i in range(len(s)):
            if s[i] == '(':
                lc += 1
            else:
                if lc > 0:
                    lc -= 1
                else:
                    extra += 1
                    
        return extra + lc
